src/main/kotlin/org/jetbrains/research/testspark/tools/llm/test/JUnitTestSuitePresenter.kt

```diff
@@ -34,16 +37,29 @@ class JUnitTestSuitePresenter(
             // Add each test
             testCases.forEach { testCase -> testBody += "$testCase\n" }
 
-            JavaClassBuilderHelper.generateCode(
-                project,
-                testFileName,
-                testBody,
-                imports,
-                packageString,
-                runWith,
-                otherInfo,
-                generatedTestsData,
-            )
+            when (language) {
+                Language.Java -> JavaClassBuilderHelper.generateCode(
+                    project,
+                    testFileName,
+                    testBody,
+                    imports,
+                    packageName,
+                    runWith,
+                    otherInfo,
+                    generatedTestsData,
+                )
+
+                Language.Kotlin -> KotlinClassBuilderHelper.generateCode(
+                    project,
+                    testFileName,
+                    testBody,
+                    imports,
+                    packageName,
+                    runWith,
+                    otherInfo,
+                    generatedTestsData,
+                )
+            }
```

Comment:
Again, if you make the `generateCode` method non-static, such constructs will be much easier to write: you would first create an instance of `ClassBuilderHelper` in a `when`-consturct, and then call the method **exactly once**.

Reply:
https://github.com/JetBrains-Research/TestSpark/issues/301